pkg/prometheus/promcfg.go

```diff
@@ -2503,7 +2503,7 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "scheme", Value: strings.ToLower(*sc.Spec.Scheme)})
 	}
 
-	cfg = cg.addProxyConfigtoYaml(ctx, cfg, sc.GetNamespace(), store, sc.Spec.ProxyConfig)
+	cfg = cg.addProxyConfigtoYaml(ctx, cfg, sc.GetNamespace(), store, &sc.Spec.ProxyConfig)
```

Comment:
(suggestion) let's pass the value directly instead of the pointer and  addProxyConfigtoYaml() can check if it's zero or not.
```suggestion
	cfg = cg.addProxyConfigtoYaml(ctx, cfg, sc.GetNamespace(), store, sc.Spec.ProxyConfig)
```